src/NuGet.Core/NuGet.Repositories/NuGetv3LocalRepository.cs

```diff
@@ -82,27 +107,10 @@ public IEnumerable<LocalPackageInfo> FindPackagesById(string packageId)
         /// </summary>
         public void ClearCacheForIds(IEnumerable<string> packageIds)
         {
-            lock (_cache)
+            foreach (var packageId in packageIds)
             {
-                foreach (var packageId in packageIds)
-                {
-                    _cache.Remove(packageId);
-                }
-            }
-        }
-
-        private IEnumerable<LocalPackageInfo> GetOrAdd(string packageId, Func<string, List<LocalPackageInfo>> factory)
-        {
-            lock (_cache)
-            {
-                IEnumerable<LocalPackageInfo> results;
-                if (!_cache.TryGetValue(packageId, out results))
-                {
-                    results = factory(packageId);
-                    _cache[packageId] = results;
-                }
-
-                return results;
+                IEnumerable<LocalPackageInfo> packages;
+                _cache.TryRemove(packageId, out packages);
```

Comment:
What does it mean for the application if `TryRemove` fails? Are we sure two concurrent `TryRemove` calls will always result in the removal from the cache?